src/chainstate/burn/db/sortdb.rs

```diff
@@ -3815,6 +3874,20 @@ impl SortitionDB {
         )
     }
 
+    /// Get the list of Peg-In operations processed in a given burnchain block.
+    /// This will be the same list in each PoX fork; it's up to the Stacks block-processing logic
+    /// to reject them.
+    pub fn get_peg_in_ops(
+        conn: &Connection,
+        burn_header_hash: &BurnchainHeaderHash,
+    ) -> Result<Vec<PegInOp>, db_error> {
+        query_rows(
+            conn,
+            "SELECT * FROM peg_in WHERE burn_header_hash = ?",
```

Comment:
Can you update the schema to add an index on `peg_in` by `burn_header_hash`?  Otherwise, this is a table scan, which leads to unacceptable performance.

Reply:
If you set `BLOCKSTACK_DB_TRACE=1` when you run your tests, the node will run and print out `EXPLAIN QUERY PLAN` on all of your queries before executing them.  You should verify from the output that there are no `SCAN TABLE` plans.

Reply:
Addressed in #3486